packages/near-membrane-base/src/membrane.ts

```diff
@@ -705,6 +720,7 @@ export function createMembraneMarshall() {
                     foreignTargetTraits,
                     foreignTargetFunctionName
                 );
+                this.isArray = !!(foreignTargetTraits & TargetTraits.IsArray);
```

Comment:
this is somehow weird, maybe storing the traits instead, and only do this operation when really needed. The reason why this is needed is because anything passing the test of isArray should be considered live, subclassing arrays is not very common, and even if that's the case, the `length` behavior and such seems like a pretty good indication that should be considered live.